### DISCUSSION TITLE
 Ansible version

### DISCUSSION THREAD

**Discussion by zelanastasia** [semaphoreui/semaphore#1266] [Q&A]
I used 5.5.0 ansible version and semaphore doesnt accept my syntax. 
Is it possible to set ansible version and how?

**Comment by jrglynn2** [comment]
I am not a developer on this project, but am a recent user. I use ansible version 7.3.0 (ansible-core 2.14.4) with no issues.

What do you mean by "doesn't accept my syntax?"

It took a little tweaking for me to get Semaphore & Ansible talking, check out issue #1252 for the problems & solution that I had...

**Reply by zelanastasia** [reply]
Maybe i don't understand how it works. Will be grateful if somebody tell me about. How semaphore use specific ansible version? 
If i run ansible with gitlab i use gitlab runner in container from ansible image and ansible-playbook runs in container and use ansible in this contianer.
How it works with semaphore?

**Reply by jrglynn2** [reply]
The way I understand it, Semaphore is a front-end that runs Ansible commands using the already-installed Ansible on the server/venv/whatever you have set up.

Besides the configuration described in the issue I linked, I also added this line to my Semaphore service file - `Environment=ANSIBLE_CONFIG=/opt/ansible/ansible.cfg` - this environment variable file points Semaphore to use my custom Ansible config file, which is where I have various configurations stored.

A greater explanation than this would have to come from one of the Semaphore contributors.

Hope this helps.

**Reply by zelanastasia** [reply]
okay, thnx, but i don't understand how to run ansible on specific ansible server ? Not on server where role should be install packages.

**Reply by jrglynn2** [reply]
That is an ansible question, not a semaphore question. I would recommend going to ansible.com and the ansible IRC chat rooms for assistance with how to use ansible.

**Reply by ansibleguy** [reply]
@zelanastasia If you have problems understanding the way Ansible works (even without Semaphore) you could read into the [Getting started Guide](https://docs.ansible.com/ansible/latest/getting_started/index.html)

After that please read the [Semaphore User Guide](https://docs.ansible-semaphore.com/user-guide)

If there is a Semaphore-Specific Problem - please provide more Information:
* what does not work
* errors that occur
* when does it happen (after you perform action xy)
* what is your semaphore configuration (task-template, repository, inventory)

What I can say: Ansible Syntax hasn't changed in a breaking way since at least 4 years now (v2.9-v7.x)
